### PR TITLE
[nnc] Started codegenning some external calls

### PR DESCRIPTION
**Pull request by Chillee**
Currently only supports native ops that have all tensor arguments, an out variant, and no kwargs.

### CODE REVIEW

File: torch/csrc/jit/tensorexpr/codegen_external.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    min_leading_spaces = min(map(num_leading_spaces, lines))
+    lines = [line[min_leading_spaces:] for line in lines]
+    return '\n'.join(lines)
+
+
+def gen_external(native_functions_path):
+    native_functions = parse_native_yaml(native_functions_path)
+    func_decls = []
+    for schema in native_functions:
+        func = schema.func
```
Comment (ljk53) [inline torch/csrc/jit/tensorexpr/codegen_external.py:18]: fyi (you can ignore), the native_functions are called functions and the 'func' field on it is actually called 'schema': https://github.com/pytorch/pytorch/blob/master/tools/codegen/model.py#L174

File: torch/csrc/jit/tensorexpr/codegen_external.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        name = func.name.name.base
+        args = func.arguments
+        # Only supports extern calls for functions with out variants
+        if func.name.overload_name != 'out':
```
Comment (ljk53) [inline torch/csrc/jit/tensorexpr/codegen_external.py:22]: nit: you probably can use `func.is_out_fn` directly: https://github.com/pytorch/pytorch/blob/master/tools/codegen/model.py#L664

**Review by ljk53 — COMMENTED** [review]
Thanks for working on this!
Please adjust the C10_MOBILE ifdef, the rest are minor comments you can ignore / do later.

File: torch/csrc/jit/tensorexpr/codegen_external.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (ezyang) [reply torch/csrc/jit/tensorexpr/codegen_external.py:1]: yes plz

[the remark this answers is not in the document]
Comment (Chillee) [reply torch/csrc/jit/tensorexpr/codegen_external.py:1]: Is there any point in doing this now? There's no downstream users of this currently, and if more ops are added it's not gonna break anybody's code.

[the remark this answers is not in the document]
Comment (ljk53) [reply torch/csrc/jit/tensorexpr/codegen_external.py:1]: > Is there any point in doing this now? There's no downstream users of this currently, and if more ops are added it's not gonna break anybody's code.

Agreed - "Not in this PR and eventually" :) I was just confirming the the desired location for these extra codegen scripts (codegen/dest folder) with ed.

File: torch/csrc/jit/tensorexpr/codegen_external.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  }} catch(...) {{
+  }}
+}}"""
+        func_registration = f"""\
+const static RegisterNNCExternalFunction nnc_{name}(
+    "nnc_aten_{name}",
+    nnc_aten_{name});"""
+        func_decls.append(func_decl)
+        func_registrations.append(func_registration)
+    external_path = "../../../../tools/jit/templates/external_functions_codegen_template.cpp"
```
Comment (ljk53) [inline torch/csrc/jit/tensorexpr/codegen_external.py:75]: Does the relative-path depend on where this script is called? (if so, you may want to consider adding a CLI argument with default value, instead.) Probably add a comment at the top of the script to show how to invoke the script?

**Review by ljk53 — APPROVED** [review]
lgtm - please fix the lint error.

File: torch/csrc/jit/tensorexpr/codegen_external.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ezyang) [reply torch/csrc/jit/tensorexpr/codegen_external.py:1]: A good reason to not check in generated code NOW is it prevents people from starting make edits to the generated code, thus making it that much more difficult to stop checking in generated code

**Pull request closed without merging**